docs/source/configuration/slots.md

```diff
@@ -0,0 +1,295 @@
+---
+myst:
+  html_meta:
+    "description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:title": "Slots"
+    "keywords": "Volto, Plone, frontend, React, configuration, slots, viewlets"
+---
+
+# Slots
+
+Slots provide a way for Volto add-ons to insert their own components at predefined locations in the rendered page.
+
+```{note}
+This concept is inspired by the Plone Classic UI {doc}`plone:classic-ui/viewlets`.
+```
+
+
+## Anatomy
+
+Slots are named, and they can contain a list of different slot components.
+Slot components are also named, and they are registered in the {ref}`configuration registry using a specific API for slots <configuration-registry-for-slot-components>`.
+
+The main trait of a slot component is that its renderer is controlled by a list of conditions called {term}`predicates`.
+Multiple slot components can be registered under the same name, as long as they have different predicates.
+
+The following tree structure diagram illustrates these concepts.
+
+```text
+Slot (`toolbar`)
+├── SlotComponent (`edit`)
+│   ├── predicate (only appear in `/de/about`)
+│   ├── predicate (only appear if the content type is either a `Document` or `News Item`)
+│   └── no predicate (default when all predicates return `false`)
+├── SlotComponent (`contents`)
+└── SlotComponent (`more`)
+```
+
+At the root of the tree, there is a slot named `toolbar`.
+It contains three slot components, named `edit`, `contents`, and `more`.
+`edit` contains three slot components, two of which have a predicate and a third without a predicate.
+When the slots with predicates both return `false`, then the third slot is returned.
+
+Thus, when either the route is `/de/about` or the content type is either a `Document` or `News Item`, then the `edit` slot component would appear in the slot `toolbar`.
+It would not display elsewhere.
```

Comment:
I don't think this paragraph is accurate. As I understand it, one of 3 SlotComponents with the name `edit` would be picked, based on which predicates match.

Reply:
@davisagli I propose to read the whole thing together this evening, what do you think @stevepiercy ?

Reply:
Yup, maybe make it a topic of the Documentation Team meeting in Discord, 1:30 AM Pacific?

Reply:
I think David's proposal below is good!